packages/grafana-ui/src/components/ThresholdsEditorNew/ThresholdsEditor.tsx

```diff
@@ -167,6 +175,7 @@ export class ThresholdsEditor extends PureComponent<Props, State> {
         key={isPercent.toString()}
         onChange={(event: ChangeEvent<HTMLInputElement>) => this.onChangeThresholdValue(event, threshold)}
         value={threshold.value}
+        ref={this.inputRef}
```

Comment:
Hm, since this is rendered in a loop, won't it create some weird edge cases when the current ref does not actually refer to the last added input?

Reply:
Yep, the behaviour you're describing is the reason why I ended up with the `focus` occurring in the `setState` callback. On mount the ref will point to the lowest editable threshold input but on clicking the add threshold button it tracks the last added and I've yet to find a scenario where it doesn't. However I'm now wondering if an alternative approach would be to keep track of every input as an array of refs and then focus the last added? What do you think?

I tried a few other approaches around using `autoFocus` and letting react do the work but they all caused varying issues when switching tabs and the tree was torn down or rebuilt.

Reply:
Ok, I see. And what about applying the ref to the topmost input only?